node/src/components/linear_chain/pending_signatures.rs

```diff
@@ -77,7 +109,33 @@ impl PendingSignatures {
         let validator_sigs = self.pending_finality_signatures.get_mut(public_key)?;
         let sig = validator_sigs.remove(&block_hash);
         self.remove_empty_entries();
-        sig
+        sig.map(|status| status.into_inner())
+    }
+
+    /// Marks pending finality signature as created by a bonded validator.
+    /// Returns whether there was a signature from the `public_key` for `block_hash` in the
+    /// collection.
+    pub(super) fn mark_bonded(&mut self, public_key: PublicKey, block_hash: BlockHash) -> bool {
+        self.mark_bonded_helper(public_key, block_hash).is_some()
+    }
+
+    fn mark_bonded_helper(&mut self, public_key: PublicKey, block_hash: BlockHash) -> Option<()> {
```

Comment:
It looks like this could be inlined into `mark_bonded` now: `Option<()>` contains no more information than a `bool`.

Reply:
Will I sill be able to call `?` to short-circuit?

Reply:
Ah, right, I hadn't realized that was the reason.